openfermioncirq/_compat_test.py

```diff
@@ -45,10 +45,17 @@ def f():
     pass
 
 
-@deprecated_test
 def test_deprecated_test():
-    warnings.simplefilter('error')
-    f()
+
+    @deprecated_test
+    def test():
+        f()
+
+    with warnings.catch_warnings(record=True) as w:
+        warnings.simplefilter('ignore')
+        warnings.simplefilter('default', DeprecationWarning)
```

Comment:
Could you verify that the test fails if you replace `DeprecationWarning` with another warning? I'm concerned that order may be important when building up the filter (and the preceding disposition is to ignore!).

Reply:
Indeed, the order does matter. Each filter is added to the front of the list of filters; see https://docs.python.org/3/library/warnings.html#warnings.filterwarnings. What I am doing here is specifying that all warnings except for DeprecationWarnings should be ignored, because I am testing specifically for the absence of a DeprecationWarning. But since I am not expecting any warnings, changing DeprecationWarning to something else shouldn't cause the test to fail. Instead, what should cause the test to fail is omitting the `deprecated_test` decorator above `test`. I have checked that that does cause the test to fail.